plugins/home/src/api/LocalStorageVisitsApi.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IdentityApi } from '@backstage/core-plugin-api';
+import { Visit } from './VisitsApi';
+import { VisitsApiFactory } from './VisitsApiFactory';
+
+/**
+ * @public
+ * This is a reference implementation of VisitsApi using window.localStorage.
+ */
+export class LocalStorageVisitsApi extends VisitsApiFactory {
+  private readonly localStorage: Window['localStorage'];
+  private readonly storageKeyPrefix = '@backstage/plugin-home:visits';
+  private readonly identityApi: IdentityApi;
+
+  constructor({
```

Comment:
This is not inline with [style conventions](https://github.com/backstage/backstage/blob/master/STYLE.md) I will fix this. To be more specific: `7.Keep constructors private, prefer static factory methods for creating instances.`